framework/sysdeps/linux/cpu_affinity.cpp

```diff
@@ -22,9 +21,22 @@ static void set_thread_name(const char *thread_name)
 
 LogicalProcessorSet ambient_logical_processor_set()
 {
-    LogicalProcessorSet result;
-    if (sched_getaffinity(0, sizeof(result.array), reinterpret_cast<cpu_set_t *>(result.array)) != 0)
-        result.clear();
+    int size = CPU_SETSIZE;
+    LogicalProcessorSet result(size);
+    int r = sched_getaffinity(0, result.size_bytes(), reinterpret_cast<cpu_set_t *>(result.array.data()));
+
+    while (r != 0 && errno == EINVAL) {
+        // increase the set size until it works
+        size *= 2;
+        result.set(LogicalProcessor(size - 1));
```

Comment:
This seems broken to me. It sets the last processor in the set and then returns it as ambient set?.. The rest of the bits are left unset -- it's probing the size, but not ambient set...

Reply:
`sched_getaffinity` will overwrite everything. This `.set` call simply enlarges the buffer. But to be clear I'll change it to `.unset`